presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/OptimizerResultCollector.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.optimizations;
+
+import com.google.common.collect.ImmutableList;
+
+import javax.annotation.concurrent.GuardedBy;
+import javax.annotation.concurrent.ThreadSafe;
+
+import java.util.LinkedList;
+import java.util.List;
+
+@ThreadSafe
```

Comment:
Why do we need threadsafe here? Every session should only correspond to 1 query. Is it needed for tests?

Reply:
it was a copy/paste from another similar class but I think you're right, we're not running optimizers in parallel so that's not needed. I'll remove it